source/common/access_log/access_log_formatter.cc

```diff
@@ -619,6 +632,36 @@ std::string DynamicMetadataFormatter::format(const Http::HeaderMap&, const Http:
   return MetadataFormatter::format(stream_info.dynamicMetadata());
 }
 
+FilterStateFormatter::FilterStateFormatter(const std::string& key,
+                                           absl::optional<size_t> max_length)
+    : key_(key), max_length_(max_length) {}
+
+std::string FilterStateFormatter::format(const Http::HeaderMap&, const Http::HeaderMap&,
+                                         const Http::HeaderMap&,
+                                         const StreamInfo::StreamInfo& stream_info) const {
+  const StreamInfo::FilterState& filter_state = stream_info.filterState();
+  if (!filter_state.hasDataWithName(key_)) {
+    return UnspecifiedValueString;
+  }
+
+  const auto& object = filter_state.getDataReadOnly<StreamInfo::FilterState::Object>(key_);
+  ProtobufTypes::MessagePtr proto = object.serializeAsProto();
+  if (proto == nullptr) {
+    return UnspecifiedValueString;
+  }
+
+  std::string value;
+  const auto status = Protobuf::util::MessageToJsonString(*proto, &value);
+  if (!status.ok()) {
+    // If the message contains an unknown Any, MessageToJsonString will fail.
+    value = proto->ShortDebugString();
```

Comment:
I'm not a huge fan of using this as a debug feature; either we should provide the value or not (or the opaque proto version). This representation is subject to change and might not be deterministic.

Reply:
OK I'll leave this as `-` and leave a TODO. protobuf doesn't allow easy fallback in such case.